comp/otelcol/converter/impl/prometheus.go

```diff
@@ -176,20 +176,46 @@ func addPrometheusReceiver(conf *confmap.Conf, promServerAddr string) {
 	onlyStaticConfigMap["targets"] = []any{promServerAddr}
 
 	addComponentToConfig(conf, comp)
-	addDDExpToInternalPipeline(conf, comp, datadogExportersMap)
+
+	processorInternalPipeline := getProcessorInternalPipeline()
+	addComponentToConfig(conf, processorInternalPipeline)
+	addDDExpToInternalPipeline(conf, []component{comp, processorInternalPipeline}, datadogExportersMap)
 }
 
-func addDDExpToInternalPipeline(conf *confmap.Conf, comp component, datadogExportersMap map[string]any) {
+func addDDExpToInternalPipeline(conf *confmap.Conf, comps []component, datadogExportersMap map[string]any) {
 	for ddExporterName := range datadogExportersMap {
 		pipelineName := "metrics" + "/" + ddAutoconfiguredSuffix + "/" + ddExporterName
-		addComponentToPipeline(conf, comp, pipelineName)
+		for _, comp := range comps {
+			addComponentToPipeline(conf, comp, pipelineName)
+		}
 		addComponentToPipeline(conf, component{
 			Type:         "exporters",
 			EnhancedName: ddExporterName,
 		}, pipelineName)
 	}
 }
 
+func getProcessorInternalPipeline() component {
+	name := "filter/drop-prometheus-internal-metrics"
+	return component{
+		Type:         "processors",
+		Name:         name,
```

Comment:
I remember this field is supposed to be just the name of the package, i.e. filter in this case